File: app/src/main/java/br/com/hussan/githubapi/adapters/DataBoundViewHolder.java
package br.com.hussan.githubapi.adapters;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;

/**
 * Created by hussan on 5/16/16.
 */

public class DataBoundViewHolder<T extends ViewDataBinding> extends RecyclerView.ViewHolder{

    private T binding;

    public DataBoundViewHolder(T binding) {
        super(binding.getRoot());
        this.binding = binding;
    }
    public T getBinding()
    {
        return binding;
    }
}

File: app/src/main/kotlin/br/com/hussan/githubapi/adapters/DataBoundViewHolder.kt
package br.com.hussan.githubapi.adapters

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView

/**
 * Created by hussan on 5/16/16.
 */

open class DataBoundViewHolder<T : ViewDataBinding>(val binding: T) : RecyclerView.ViewHolder(binding.root)
